File: backend/entity/Application.go
package entity

import (
	"time"

	"gorm.io/gorm"
)

type Application struct {
	gorm.Model
	Status        string `json:"status"`
	ResumeUrl     string `json:"resume_url"`
	TranscriptUrl string
	SubmitAt      time.Time `json:"submit_at"`
	CompanyNote   string    `json:"company_note"`

	IntershipPostID uint          // FK to Intership_post
	IntershipPost   IntershipPost `gorm:"foreignKey:IntershipPostID"`

	ApplicationDetails []ApplicationDetails `gorm:"foreignKey:ApplicationID"` // one-to-many
}

File: frontend/src/interfaces/Application.ts
import type { InternshipPostInterface } from "./InternshipPost";
import type { ApplicationDetailInterface } from "./ApplicationDetail";

export interface ApplicationInterface {
  id?: number;
  status?: string;
  resume_url?: string;
  submit_at?: string;

  intership_post_id?: number;
  intership_post?: InternshipPostInterface;

  application_details?: ApplicationDetailInterface[];
}
